test/cloudformation/cloudformation_test.go

```diff
@@ -0,0 +1,98 @@
+package cloudformation
+
+import (
+	"context"
+	"flag"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	timeOutMinutes = 15
+	instanceIdKey  = "InstanceId"
+	namespace      = "CFMetricTest"
+)
+
+var (
+	dimensionProviders = []dimension.IProvider{
+		&dimension.CustomDimensionProvider{Provider: dimension.Provider{}},
+	}
+	dimensionFactory = dimension.Factory{Providers: dimensionProviders}
+	packagePath      = flag.String("package_path", "", "s3 path to download package")
+	iamRole          = flag.String("iam_role", "", "iam role for cf ec2 instance")
+	keyName          = flag.String("key_name", "", "key name for ec2 instance")
+	metricName       = flag.String("metric_name", "", "metric to look for")
+)
+
+func TestCloudformation(t *testing.T) {
+	log.Printf("Package path %s iam role %s key name %s metric name %s", *packagePath, *iamRole, *keyName, *metricName)
+	ctx := context.Background()
+	stackName := awsservice.CreateStackName("cfTestStack")
+
+	client := awsservice.CloudformationClient
+
+	// Read template file
+	template, err := os.ReadFile("resources/AmazonCloudWatchAgent/inline/amazon_linux.template")
```

Comment:
I commented in the other PR, but I don't like that we're taking a dependency on a CloudFormation template that is hosted in a repo we have no control over. We should just drop in a CFT that we want to use into this package and use that.